src/dectris/neggia/plugin/H5ToXds.cpp

```diff
@@ -55,8 +55,7 @@ void printVersionInfo() {
    std::cout << "This is neggia " << VERSION << " (Copyright Dectris 2017)" << std::endl;
 }
 
-template<class T> void applyMaskAndTransformToInt32(const T * indata, int outdata[], const uint32_t * maskData, size_t size) {
-    constexpr size_t maxSigned = (size_t)std::numeric_limits<int32_t>::max();
+template<class T> void applyMaskAndTransformToInt32(const T * indata, int outdata[], const uint32_t * maskData, size_t size, size_t maxSigned) {
```

Comment:
simpler:
```constexpr size_t maxSigned = (size_t)std::numeric_limits<std::make_signed<T>::type>::max();```

Reply:
Absolutely: I'm no C++ expert by a (very) long stretch ... The main idea was to show you what I did to resolve the 16/32-bit issue and you can take the lead in providing the "proper" solution here.